File: redsheet-portal/src/app/database-models/batna-detail.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { Ourbatna, TheirBatna,  } from './'
export class BatnaDetail {
    constructor(batnaDetail?: BatnaDetail )  {
        let properties = [ "batnaDetailId", "batnaDetailValue", "ourbatnas", "theirBatnas",];
        for (let property of properties)
            if (batnaDetail && batnaDetail[property])
                this[property] = batnaDetail[property];
    }
 
	batnaDetailId : number =   0 ;
 
    @required()
    @maxLength(1000)
	batnaDetailValue : string =   undefined;
	@nested(Ourbatna)
	ourbatnas: Ourbatna[];

	@nested(TheirBatna)
	theirBatnas: TheirBatna[];



}

File: RedSheet.DbEntities/Models/BatnaDetail.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("BatnaDetails", Schema = "dbo")]
	[Serializable]
	public partial class BatnaDetail  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int BatnaDetailId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string BatnaDetailValue { get; set; } 
	[InverseProperty("BatnaDetail")]
	public ICollection<Ourbatna> Ourbatnas { get; set; } 

	[InverseProperty("BatnaDetail")]
	public ICollection<TheirBatna> TheirBatnas { get; set; } 


	public BatnaDetail()
	{
	this.Ourbatnas = new HashSet<Ourbatna>();

	this.TheirBatnas = new HashSet<TheirBatna>();

    }

    }
}

